### PR TITLE
Dkh task event states

### PR DESCRIPTION
**Pull request by dkhuntrods**
I want to begin merging this now; not all display text is 100% accurate, and there's probably some refactoring needed, but everything is functional and the rest is just minor tweaks.

### CODE REVIEW

File: blocks/core/ff_module/ff_module-task-event/_src/taskEventWithOptionalMessage.js [hunk 1]
```diff
@@ -1,17 +1,67 @@
...
 
-        return  <TaskEventBase description={this.props.description}>
+        return  <TaskEventBase
+                    description={this.props.description}
+                    actions={this.props.actions}
+                    state={this.props.state}
+                    onRetryAfterStatusError={this.props.onRetryAfterStatusError}>
+                    <p className='ff_module-task-event__author-action'>{outputText}</p>
+                    {message}
+                </TaskEventBase>;
```
Comment (bwfox) [inline blocks/core/ff_module/ff_module-task-event/_src/taskEventWithOptionalMessage.js]: Some duplication here between the rendering of the different states. Could extract this out into a function that takes `props`, `outputText` and `message` as parameters.

**Comment by bwfox** [comment]
Left a comment about one minor thing. Other than that, LGTM.

File: blocks/core/ff_module/ff_module-task-event/_src/taskEventWithOptionalMessage.js (hunk 1 above)
Comment (dkhuntrods) [inline blocks/core/ff_module/ff_module-task-event/_src/taskEventWithOptionalMessage.js]: Yes, agreed, will tackle that. Thanks.

**Comment by bwfox** [comment]
@dkhuntrods I am wondering if instead of having `Sent` and `Saved` states, we want just one of those, coupled with the `released` state. So `sent` would be `released`, `saved` would be `!released`. This would applied in a similar way for the `pending` and `error` states. E.g., `error` + `released` is `errorSending`, `error` + `!released` is `errorSaving`. Etc.

I think that would also make the front-end logic simpler as the pending, saved, error part of the state when saving is independent from the released/unreleased part.

**Comment by dkhuntrods** [comment]
This is done; just waiting for webapp to merge the branch with supporting changes before merging into dev.

**Pull request merged**